Translated the "How to contribute" page (#93)
reactjs/fr.reactjs.org: content/docs/how-to-contribute.md

Tout travail sur React se passe directement sur [GitHub](https://github.com/facebook/react). Les membres de l'équipe principale (NdT core team) aussi bien que les contributeurs externes y envoient leur pull requests qui passent à travers le même processus de revue.

### Organisation des branches {#branch-organization}

Nous ferons de notre mieux pour garder la branche [`master`](https://github.com/facebook/react/tree/master) en bonne condition, avec les tests qui passent en tout temps. Mais pour pouvoir avancer rapidement, nous ferons des changements d'API avec lesquels votre application pourrait ne pas être compatible. Nous vous recommandons d'utiliser [la dernière version stable de React](/downloads.html).

Si vous envoyez une pull request, merci de la faire vers la branche `master`. Nous maintenons des branches  stables pour chaque version majeure séparément mais n'acceptons pas de pull requests vers ces dernières directement. A  la place, nous faisons un cherry-pick des changements qui ne créent pas de rupture (NdT non-breaking) de master à la version stable majeure la plus récente.

### Gestion de versions sémantique {#semantic-versioning}

> 

Author: Désolé 😰

### Envoyer une pull request {#sending-a-pull-request}

L'équipe principale surveille les pull requests. Nous faisons une revue de votre pull request et soit nous la fusionnerons, soit nous demanderons  des changements, soit nous la fermerons avec une explication. Pour les changements d'API, nous aurions peut-être besoin de changer nos usages internes à facebook.com, ce qui pourrait retarder la fusion. Nous ferons cependant notre mieux pour donner des nouvelles et des retours pendant le processus.

**Avant de soumettre une pull request,** veuillez faire attention à suivre les instructions suivantes :

1. Forkez [le dépôt](https://github.com/facebook/react) et créez votre branche depuis `master`.
2. Lancez `yarn`  à la racine du dépôt.
3. Si vous avez réparé un bug ou ajouté  du code qui devrait être testé, ajouter des tests !
4. Assurez-vous que les suites de test passent (`yarn test`). Astuce : `yarn test --watch NomDuTest` est très utile en phase de développement.
5. Lancez `yarn test-prod` pour tester dans l'environnement de production. Cette commande accepte les même options que `yarn test`.

> 6. Si vous avez besoin d'un debugger, lancez `yarn debug-test --watch NomDuTest`, ouvrez `chrome://inspect`, et appuyez sur “Inspecter”.

Reviewer: ```suggestion
6. Si vous avez besoin d'un débogueur, lancez `yarn debug-test --watch NomDuTest`, ouvrez `chrome://inspect`, et appuyez sur « Inspecter ».
```

7. Formattez votre code avec [prettier](https://github.com/prettier/prettier) (`yarn prettier`).

> 8. Assurez-vous que votre code passe la vérification du linter (`yarn lint`). Astuce : `yarn linc` pour ne vérifier que les fichiers qui ont été changés.

Reviewer: ```suggestion
8. Assurez-vous que votre code passe la vérification du linter (`yarn lint`). Astuce : `yarn linc` ne vérifiera que les fichiers qui ont changé.
```

9. Lancez les vérifications de types [Flow](https://flowtype.org/) (`yarn flow`).

> 10. Si vous ne l'avez pas encore fait, complétez le CLA.

Reviewer: ```suggestion
10. Si vous ne l'avez pas encore fait, remplissez le CLA (voir ci-dessous).
```

> ***************

Reviewer: ```suggestion
```

Mais d'où tu rajoutes une barre de séparation ? Ça remet le boxon dans le diff 😢

> ### Accord de license de contributeur (NdT Contributor License Agreement) (CLA) {#contributor-license-agreement-cla}

Reviewer: ```suggestion
### Accord de licence de contribution (CLA) {#contributor-license-agreement-cla}
```

La NdT sera reportée ensuite, et « contribution » permet de rester inclusif sans alourdir le titre avec une écriture inclusive un peu longue sur ce coup.

> Pour que nous puissions accepter votre pull request, il est nécessaire que vous soumettez un CLA. Vous ne devez faire cela qu'une seule fois, donc si vous l'avez déjà fait pour un autre projet open source de Facebook, pas besoin de le refaire. Si vous soumettez une pull request pour la première fois, dites-nous simplement que vous avez déjà complété le CLA et nous pourrons vérifier cela avec votre nom d'utilisateur GitHub.

Reviewer: In order to accept your pull request, we need you to submit a CLA. You only need to do this once, so if you've done this for another Facebook open source project, you're good to go. If you are submitting a pull request for the first time, just let us know that you have completed the CLA and we can cross-check with your GitHub username.

```suggestion
Afin que nous puissions accepter votre _pull request_, nous avons besoin que vous soumettiez un CLA *(Contributor License Agreement, NdT)*. Vous n’avez besoin de faire ça qu'une seule fois, donc si vous l'avez déjà fait pour un autre projet open source de Facebook, tout va bien. Si vous envoyez une _pull request_ pour la première fois, dites-nous simplement que vous avez déjà rempli le CLA et nous pourrons le vérifier sur base de votre identifiant GitHub.
```

> **[Complétez votre CLA ici.](https://code.facebook.com/cla)**

Reviewer: ```suggestion
**[Remplissez votre CLA ici.](https://code.facebook.com/cla)**
```

### Pré-requis pour contribuer {#contribution-prerequisites}

* Vous avez [Node](https://nodejs.org) installé en v8.0.0+ et [Yarn](https://yarnpkg.com/en/) en v1.2.0+.

> * Vous avez `gcc` installé ou êtes à l'aise avec le fait d'installer un compilateur si besoin. Certaines de nos dépendances peuvent nécessiter une étape de compilation. Sur OS X, les outils de ligne de commande XCode s'occupent de cela. Sur Ubuntu, `apt-get install build-essential` installera les paquets nécessaires. Des commandes similaires devraient fonctionner sur d'autres distributions Linux. Windows nécessite quelques étapes additionelles, voir les [instructions d'installation de `node-gyp`]https://github.com/nodejs/node-gyp#installation) pour plus de détails.

Reviewer: ```suggestion
* Vous avez `gcc` installé ou êtes à l'aise avec le fait d'installer un compilateur si besoin. Certaines de nos dépendances peuvent nécessiter une étape de compilation. Sur OS X, les outils de ligne de commande XCode s'en occupent. Sur Ubuntu, `apt-get install build-essential` installera les paquets nécessaires. Des commandes similaires devraient fonctionner pour d'autres distributions Linux. Windows nécessite quelques étapes supplémentaires, consultez les [instructions d'installation de `node-gyp`]https://github.com/nodejs/node-gyp#installation) pour plus de détails.
```

* Vous êtes à l’aise avec Git

### Workflow de développement  {#development-workflow}

Après avoir cloné React, lancez `yarn` afin d’aller chercher ses dépendances.
Ensuite, vous pouvez lancer différentes commandes :

* `yarn lint` pour vérifier le style du code.

> * `yarn linc` est comme `yarn lint` mais va plus vite car elle ne vérifie que les fichiers qui ont été changés sur votre branche.

Reviewer: ```suggestion
* `yarn linc` est comme `yarn lint` mais va plus vite car elle ne vérifie que les fichiers qui ont changé sur votre branche.
```

> * `yarn test` lance la suite de testa complète.

Reviewer: ```suggestion
* `yarn test` lance la suite de tests complète.
```

> * `yarn test --watch` lance un observateur de test interactif.

Reviewer: ```suggestion
* `yarn test --watch` lance un superviseur interactif de tests.
```

> * `yarn test <schema>` lance les tests des fichiers au nokm correspondant au schéma.

Reviewer: ```suggestion
* `yarn test <motif>` lance les tests des fichiers dont le nom correspond au motif.
```

> * `yarn test-prod` lance les tests dans l’environnement de production. Elle supporte toutes les mêmes options que `yarn test`.

Reviewer: ```suggestion
* `yarn test-prod` lance les tests dans l’environnement de production. Elle accepte toutes les mêmes options que `yarn test`.
```

> * `yarn debug-test` est exactement comme `yarn test` mais avec un debugger. Ouvrez `chrome://inspect` et appuyez sur “Inspecter”.

Reviewer: ```suggestion
* `yarn debug-test` est exactement comme `yarn test` mais avec un débogueur. Ouvrez `chrome://inspect` et appuyez sur « Inspecter ».
```

* `yarn flow` lance les vérifications de types [Flow](https://flowtype.org/).

> * `yarn build` créée un dossier `build` avec tous les paquets.

Reviewer: ```suggestion
* `yarn build` crée un dossier `build` avec tous les modules.
```

> * `yarn build react/index,react-dom/index --type=UMD` créée des builds UMD seulement de React et ReactDOM.

Reviewer: ```suggestion
* `yarn build react/index,react-dom/index --type=UMD` crée des builds UMD seulement de React et ReactDOM.
```

> Nous recommandons d'utiliser `yarn test` (ou ses variations mentionnées ci-dessus) pour vous assurer de ne pas introduire de régressions en travaillant sur votre changement. Cependant, il peut être pratique d'essayer votre build de React dans un vrai projet.

Reviewer: ```suggestion
Nous recommandons d'utiliser `yarn test` (ou ses variations mentionnées ci-dessus) pour vous assurer de ne pas introduire de régressions en travaillant sur votre contribution. Cependant, il peut être pratique d'essayer votre build de React dans un vrai projet.
```

« changement » c'est vraiment pas le top en français pour ce type de truc.

> 

Reviewer: T'as vraiment un truc avec les doubles sauts de ligne, dis donc…

> Tout d’abord, lancez `yarn build`. Cela produira des bundles pré-compilés dans le dossier `build`, de même que préparer les paquets npm dans `build/packages`.

Reviewer: ```suggestion
Tout d’abord, lancez `yarn build`. Ça produira des _bundles_ pré-compilés dans le dossier `build`, et préparera les modules npm dans `build/packages`.
```

> La façon la plus facile d'essayer vos changements est de lancer `yarn build react/index,react-dom/index --type=UMD` et ensuite d’ouvrir `fixtures/packaging/babel-standalone/dev.html`. Ce fichier utilise déjà `react.development.js` du dossier `build`, donc il utilisera vos changements.

Reviewer: ```suggestion
La manière la plus simple d'essayer vos modifications consiste à lancer `yarn build react/index,react-dom/index --type=UMD` et ensuite ouvrir `fixtures/packaging/babel-standalone/dev.html`. Ce fichier utilise déjà `react.development.js` depuis le dossier `build`, donc il utilisera vos évolutions.
```

On évite les propositions sujet de « est », en préférant « consiste à » le plus souvent.

> Si vous voulez essayer vos changements dans votre projet React existant, vous pouvez copier `build/dist/react.development.js`, `build/dist/react-dom.development.js`, ou tout autre produits de la compilation dans votre appli et utilisez les à la place de la version stable. Si votre projet utilise React depuis npm, vous pouvez supprimer `react` et `react-dom` dans ses dépendances et utiliser `yarn link` pour les pointer vers votre dossier local `build` :

Reviewer: ```suggestion
Si vous voulez essayer vos évolutions dans votre projet React existant, vous pouvez copier `build/dist/react.development.js`, `build/dist/react-dom.development.js`, ou tout autre produit de la compilation dans votre appli et les utiliser au lieu de la version stable. Si votre projet utilise React depuis npm, vous pouvez supprimer `react` et `react-dom` dans ses dépendances et utiliser `yarn link` pour les faire pointer vers votre dossier local `build` :
```

On évite « à la place » (cf. Glossaire)

```sh
cd ~/chemin_vers_votre_clone_de_react/build/node_modules/react
yarn link
cd ~/chemin_vers_votre_clone_de_react/build/node_modules/react-dom
yarn link
cd /chemin/vers/votre/projet
yarn link react react-dom
```

> Chaque fois que vous lancez `yarn build` dans le dossier `react`, les versions mises à jour apparaîtront dans le dossier `node_modules` de votre projet. Vous pouvez alors re-compiler votre projet pour essayer vos changements.

Reviewer: ```suggestion
Chaque fois que vous lancez `yarn build` dans le dossier de React, les versions mises à jour apparaîtront dans le dossier `node_modules` de votre projet. Vous pouvez alors recompiler votre projet pour essayer vos modifications.
```

On parle du dossier du clone local de ton fork, pas d'un dossier nécessaire appelé `react` (ce n'est que le nom par défaut).

> Nous demandons tout de même que votre pull request contienne des tests unitaires pour chaque nouvelle fonctionnalité. De cette manière, nous pouvons nous assurer de ne pas briser votre code par la suite.

Reviewer: ```suggestion
Nous exigeons tout de même que votre _pull request_ contienne des tests unitaires pour chaque nouvelle fonctionnalité. Ainsi nous pouvons nous assurer de ne pas casser votre code par la suite.
```

« briser » c'est un peu mélodramatique, et surtout pas très répandu pour ce cas de figure

### Guide de style {#style-guide}

> 

Reviewer: Et bim, encore un double saut de ligne qui tue le diff… J'aurais tellement dû faire un commit de recalage **avant** de lancer ma revue, c'est bien trop tard maintenant 😢

Nous utilisons un outil de formatage automatique qui s'appelle [Prettier](https://prettier.io/).

> Lancez `yarn prettier` après avoir fait un changement du code.

Reviewer: ```suggestion
Lancez `yarn prettier` après avoir changé le code.
```

> Ensuite, notre linter attrappera la plupart des problèmes qui pourraient exister dans votre code.

Reviewer: ```suggestion
Ensuite, notre linter attrapera la plupart des problèmes qui pourraient exister dans votre code.
```

Vous pouvez vérifier l’état du style de votre code simplement en lançant `yarn linc`.

> Cependant, il y a toujours certains styles que le linter ne peut pas remarquer. Si vous n'êtes pas sûr de quelque chose, laissez vous guider par le [Guide de style de Airbnb](https://github.com/airbnb/javascript).

Reviewer: ```suggestion
Cependant, il y a toujours certains styles que le linter ne peut pas remarquer. Si vous n'êtes pas sûr·e de votre coup, laissez-vous guider par le [Guide de style de Airbnb](https://github.com/airbnb/javascript).
```

> ### Vidéo d’introduction {#introductory-video}

Reviewer: ```suggestion
### Vidéo de présentation {#introductory-video}
```

> Vous pourriez être intéressé par cette [cette courte vidéo](https://www.youtube.com/watch?v=wUpPsEcGsg8) (26 mins) qui fait une introduction sur comment contribuer à React.

Reviewer: ```suggestion
Vous voudrez peut-être regarder cette [cette courte vidéo](https://www.youtube.com/watch?v=wUpPsEcGsg8) (26 mins) (en anglais) qui présente comment contribuer à React.
```

> #### Points d'intérets de la vidéo : {#video-highlights}

Reviewer: ```suggestion
#### Moments les plus intéressants de la vidéo : {#video-highlights}
```

- [4:12](https://youtu.be/wUpPsEcGsg8?t=4m12s) - Compiler et tester React en local

> - [6:07](https://youtu.be/wUpPsEcGsg8?t=6m7s) - Créer et soumettre des pull requests

Reviewer: ```suggestion
- [6:07](https://youtu.be/wUpPsEcGsg8?t=6m7s) - Créer et envoyer des _pull requests_
```

- [8:25](https://youtu.be/wUpPsEcGsg8?t=8m25s) - Organiser son code
- [14:43](https://youtu.be/wUpPsEcGsg8?t=14m43s) - npm & React
- [19:15](https://youtu.be/wUpPsEcGsg8?t=19m15s) - Ajouter de nouvelles fonctionnalités à React

> Pour avoir une vue d'ensemble réaliste de l’expérience de contribuer à React pour la première fois, jetez un oeil à [cette présentation divertissante de ReactNYC](https://www.youtube.com/watch?v=GWCcZ6fnpn4).

Reviewer: ```suggestion
Pour avoir une vue d'ensemble réaliste de l’expérience de première contribution à React, jetez un coup d’œil à [cette présentation divertissante de ReactNYC](https://www.youtube.com/watch?v=GWCcZ6fnpn4) (en anglais).
```

> ### Requête de commentaires (RFC) {#request-for-comments-rfc}

Reviewer: ```suggestion
### Appels à commentaires (RFC) {#request-for-comments-rfc}
```

> Beaucoup de changements, y compris les réparations de bugs et les améliorations de documentation, peuvent être implémentés et revus par le workflow normal de pull requests GitHub.

Reviewer: ```suggestion
Beaucoup de modifications, y compris les correctifs de bugs et les améliorations de la documentation, peuvent être implémentés et revus via le workflow normal de _pull requests_ sur GitHub.
```

> Certains changements sont cependant plus “substantiels”, et nous demandons que ceux-ci passent par une petite phase de conception pour créer un consensus au sein de l’équipe principale de React.

Reviewer: ```suggestion
Certaines évolutions sont cependant plus « substantielles », et nous demandons à ce que celles-ci passent par une petite phase de conception afin d’obtenir un consensus au sein de l’équipe noyau de React.
```

> Le processus de “RFC” (requête de commentaires) a pour but de fournir un processus controllé et cohérent pour l'introduction de nouvelles fonctionnalités dans le projet. Vous pouvez apporter votre contribution en consultant le [dépôt des rfcs](https://github.com/reactjs/rfcs).

Reviewer: ```suggestion
Le processus de « RFC » *(Request For Comments, NdT)* a pour but de fournir un chemin contrôlé et cohérent pour l'introduction de nouvelles fonctionnalités dans le projet. Vous pouvez apporter votre contribution en consultant le [dépôt des RFC](https://github.com/reactjs/rfcs).
```

- La NdT trouve sa place ici
- « Chemin » est plus juste (et fidèle) que « processus » ici
- On ne pluralise pas les acronymes en français

### Licence {#license}

En contribuant à React, vous acceptez que toute contribution soit licenciée sous la licence MIT.

> ### Quelle est la suite ? {#what-next}

Reviewer: ```suggestion
### Et maintenant ? {#what-next}
```

Trad normalisée

> Lisez la [section suivante](/docs/codebase-overview.html) pour apprendre comment la base de code est organisée.

Reviewer: ```suggestion
Lisez la [page suivante](/docs/codebase-overview.html) pour apprendre comment la base de code est organisée.
```

« Section » sous-entend la page courante en français
